### PR TITLE
fix: SKFP-1155 update mailchimp service response and error handling

### PR DESCRIPTION
**Pull request by aperron-ferlab**

### CODE REVIEW

File: src/endpoints/subscribe.ts [hunk 1]
```diff
@@ -8,14 +8,14 @@ export default (mailchimpSecret: MailChimpSecrets) =>
...
-                await newMailchimpSubscription(user, mailchimpSecret);
+                const subscriptionResponses = await newMailchimpSubscription(user, mailchimpSecret);
+                res.status(200).json({ message: 'Subscription successful', subscriptions: subscriptionResponses });
+            } else {
+                res.status(200).json({ message: 'No subscription needed' });
             }
-            res.end();
         } catch (err) {
-            console.error('subscribe route ', err);
+            console.error('Subscribe route error:', err);
```
Comment (evans-g-crsj) [inline src/endpoints/subscribe.ts:18]: is it useful to append `Subcribe...`? The error stack should contain the line where the error had occurred.

File: src/services/mailChimp.ts [hunk 2]
```diff
@@ -2,38 +2,75 @@ import fetch from 'node-fetch';
...
+                    throw new Error(responseData.detail || response.statusText);
+                }
+
+                return { subscription: subscription.type, status: 'subscribed', detail: responseData.detail };
+            }),
+        );
+
+        return responses;
+    } catch (error) {
+        console.error('An error occurred while subscribing to Mailchimp:', error);
```
Comment (evans-g-crsj) [inline src/services/mailChimp.ts:73]: same comment as above

File: src/services/mailChimp.ts [hunk 3]
```diff
@@ -2,38 +2,75 @@ import fetch from 'node-fetch';
...
+    }
+
+    if (user.acceptedKfOptIn) {
+        subscriptions.push({
+            type: 'Newsletter Subscription',
+            url: `https://${mailChimpDataCenter}.api.mailchimp.com/3.0/lists/${kfMailchimpListId}/members/`,
+        });
+    }
+
+    try {
```
Comment (evans-g-crsj) [inline src/services/mailChimp.ts:41]: Personally, I would use a for-loop

**Review by evans-g-crsj — APPROVED** [review]

File: src/services/mailChimp.ts (hunk 3 above)
Comment (aperron-ferlab) [reply src/services/mailChimp.ts:41]: This was already in place and tbh we will soon-ish move to Users API, i<d rather not touch the subscription logic

**Pull request merged**